website/static/js/accountSettings.js

```diff
@@ -0,0 +1,187 @@
+'use strict';
+
+var $ = require('jquery');
+var $osf = require('js/osfHelpers');
+var ko = require('knockout');
+var oop = require('js/oop');
+var Raven = require('raven-js');
+
+require('knockout.punches');
+ko.punches.enableAll();
+
+
+var UserEmail = oop.defclass({
+    constructor: function() {
+        this.address = ko.observable();
+        this.isConfirmed = ko.observable();
+        this.isPrimary = ko.observable();
+    }
+});
+
+
+var UserProfile = oop.defclass({
+    constructor: function () {
+
+        this.id = ko.observable();
+        this.emails = ko.observableArray();
+
+        this.primaryEmail = ko.computed(function () {
```

Comment:
Are there guidelines yet on when to use a `pureComputed` vs. `computed`? I've read the Knockout docs and I think I grok when to use it, but I'd like to bounce it off you in person before I'm confident of my understanding.